src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1187,6 +1200,12 @@ internal async Task Cleanup(bool async, bool connectionClosing = false, bool isD
             }
         }
 
+        if (ColumnInfoCache is { } cache)
```

Comment:
Should we want to recycle this across reader reuses, i.e. only return the cache to the pool when the connector is closed?

Reply:
Could, I didn't really see this come up much beyond a minuscule attribution during the profiling I did.